src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamespaceSymbol.cs

```diff
@@ -42,6 +42,14 @@ internal abstract class PENamespaceSymbol
         /// </summary>
         private Dictionary<string, TypeDefinitionHandle> _lazyNoPiaLocalTypes;
 
+        /// <summary>
+        /// Caches the last call to GetMembers(string) when that member is a namespace. Multiple types
+        /// within the same namespace are often requested in sequence, so this allows the second and
+        /// subsequent request for members with the same name to share a result without looking up the
+        /// values or allocating.
```

Comment:
I find this description of the motivation very confusing. For example, "Multiple types within the same namespace are often requested in sequence" - not sure how this is related to caching a namespace result. This "so this allows the second and subsequent request for members with the same name to share a result without looking up the values or allocating" doesn't look accurate because it sounds like the optimization is unconditional, but it is.